project/simpson0114/README.rst

```diff
@@ -0,0 +1,66 @@
+==================================================
+Minimum Spanning Tree Algorithm
+==================================================
+
+Basic Information
+=================
+**Minimum Spanning Tree(MST)** is a significant graph algorithm structure. In a connected, edge-weighted undirected graph, 
+a minimum spanning tree is a subset of edges that connects all nodes with the smallest total weights and has been well-studied for nearly a century.
+
+Problem to Solve
+=================
+In the project we try to make MST faster and easier
+
+MST is a classic graph algorithm, and it is useful to many problems in real life. However implement a MST algorithm is not a walk in the park, 
+we hope to build a framework that user just have to decide the input graph vertex and edge number, and the range of the edges weights.
+
+There are two famous Minimum spanning tree (MST) algorithms, which are Prim's algorithm and Kruskal's algorithm respectively. We will implement 
+this two MST algorithms, and hope it can make it easier for beginners to understand this two type of MST algorithms.
+
+Prospective Users
+==================
+People who wants to use MST but have no idea of programming or people who just start to learn the MST.
+
+System Architecture
+====================
+
+.. image:: images/MST_system_architecture.png
+
+API Description
+=================
+
+Graph class includes complete implementation of the Kruskal' algorithm and Prim's algorithm, user only
+have to initialize Graph in python script and choose the MST algorithm type.
+
+- Graph.__init__(): initialize vertices 
+- Graph.addEdge(): add an edge to graph
+- Graph.find(): find set of an element i
+- Graph.union(): union of two sets of x and y
+- Graph.minKey(): A utility function to find the vertex with minimum distance value, from the set of vertices not yet included in shortest path tree
+- Graph.sort(): sort edges' weight of the Graph from small to large
+- Graph.KruskalMST(): The main function to construct MST using Kruskal's algorithm
+- Graph.PrimMST(): The main function to construct MST using Prim's algorithm
+
+Engineering Infrastructure
+===========================
+1. Automatic build system and how to build your program: GNU make
+2. Version control (show how you will use it): Git
+3. Testing framework: Pytest
+
+Schedule
+=========
+* Planning phase (6 weeks including 2/21, 2/28, 3/7, 3/14, 3/21, 3/28):
+* Week 1 (4/4): Complete pure python version of MST class. Start the presentation skeleton.
+* Week 2 (4/11): Design of data structures of C++
```

Comment:
It is correct to start with data structure design.  You may also need to use some of the operations to drive the design.

Reply:
Does it mean tell more about data structure design?
Because i cannot accurately predict what problems i will meet as implementation.

Reply:
You may think and describe more about the data structures you will use, but it does _not_ mean you need to decide every detail about it.  Instead, you should list what may or may not change along the course of your code development.